ApiDoctor.Validation/Config/DocumentOutlineFile.cs

```diff
@@ -1,93 +1,177 @@
+                    ExpectedDocumentHeader expectedDocHeader => new ExpectedDocumentHeader(expectedDocHeader),
+                    _ => throw new InvalidOperationException("Unexpected header type")
+                });
             }
+            return headersCopy;
         }
     }
 
-    public class DocumentHeader
+    public class ConditionalDocumentHeader
     {
-        public DocumentHeader()
+        [JsonProperty("condition")]
+        public string Condition { get; set; }
+
+        [JsonProperty("arguments"), JsonConverter(typeof(DocumentHeaderJsonConverter))]
+        public List<object> Arguments { get; set; }
+
+        public ConditionalOperator? Operator
         {
-            Level = 1;
-            ChildHeaders = new List<DocumentHeader>();
+            get
+            {
+                return Enum.TryParse(this.Condition, true, out ConditionalOperator op) ? op : null;
+            }
         }
 
-        /// <summary>
-        /// Represents the header level using markdown formatting (1=#, 2=##, 3=###, 4=####, 5=#####, 6=######)
-        /// </summary>
-        [JsonProperty("level")]
-        public int Level { get; set; }
+        public ConditionalDocumentHeader(ConditionalDocumentHeader original)
+        {
+           if (original == null)
+                return;
 
-        /// <summary>
-        /// Indicates that a header at this level is required.
-        /// </summary>
-        [JsonProperty("required")]
-        public bool Required { get; set; }
+           Condition = original.Condition;
 
-        /// <summary>
-        /// The expected value of a title or empty to indicate any value
-        /// </summary>
-        [JsonProperty("title")]
-        public string Title { get; set; }
+           Arguments = ExpectedDocumentHeader.CopyHeaders(original.Arguments);
+        }
+    }
 
-        /// <summary>
-        /// Specifies the headers that are allowed under this header.
-        /// </summary>
-        [JsonProperty("headers")]
-        public List<DocumentHeader> ChildHeaders { get; set; }
+    public enum ConditionalOperator
+    {
+        OR,
+        AND
+    }
+
+    public class DocumentHeaderJsonConverter : JsonConverter
+    {
+        public override bool CanConvert(Type objectType)
+        {
+            return objectType == typeof(ExpectedDocumentHeader) || objectType == typeof(ConditionalDocumentHeader);
+        }
 
-        internal bool Matches(DocumentHeader found)
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
         {
-            return this.Level == found.Level && DoTitlesMatch(this.Title, found.Title);
+            if (reader.TokenType == JsonToken.StartArray)
+            {
+                var allowedHeaders = new List<object>();
+                var jArray = JArray.Load(reader);
+                foreach (var item in jArray)
+                {
+                    if (item["condition"] != null)
```

Comment:
```suggestion
                    if (item.ContainsKey("condition"))
```

Reply:
`item` is of type `JToken` and not dictionary.  Using `ContainsKey` will not work.

Reply:
isn't it a JObject? couldn't we add a type assertion?

Reply:
It's a `JToken`. What are your reservations about performing the check as I am?

Reply:
I think @baywet means we can do something like this to verify that the `JToken` is a `JObject`
```cs
if (item is not JObject jobject)
```
Then use the `ContainsKey` from JObject. as indexing may throw an key not found error if I'm not wrong.

Reply:
Thanks @andrueastman. I've made the update.